### PR TITLE
fix(openapi): do not use jsonld schema for all media types

### LINKED ISSUE #4215: After upgrade to 2.6.4 - swagger schema uses jsonld for all media types
**API Platform version(s) affected**: 2.6.4 (regression from 2.6.3)

**Description**  
(extracted from #4212)
The generated OpenAPI docs use the same schemas for content mime types `application/json` (and `application/html`) as for `application/ld+json`

**How to reproduce**  
Default config, Dummy resource (id, name)

**Possible Solution**  
After #4213, move both occurrences of
```php
            $schema = new Schema('openapi');
            $schema->setDefinitions($schemas);
```
to inside respectively
`foreach ($responseMimeTypes as $operationFormat) {` and
`foreach ($requestMimeTypes as $operationFormat) {`
(just before each call to `$this->jsonSchemaFactory->buildSchema(/*...*/)`)

**Additional Context**  

(the bug is not in Swagger-UI but really in the generated docs.json)

### PR DESCRIPTION
**Pull request by guilliamxavier**
| Q             | A
| ------------- | ---
| Branch?       | 2.6
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #4215
| License       | MIT
| Doc PR        | 

Follows #4213

### CODE REVIEW

File: features/openapi/docs.feature [hunk 1]
```diff
@@ -109,6 +109,103 @@ Feature: Documentation support
...
+    """
+    {
+        "schema": {
+            "type": "array",
+            "items": {
+                "$ref": "#/components/schemas/Dummy"
+            }
+        }
+    }
+    """
```
Comment (guilliamxavier) [inline features/openapi/docs.feature:207]: This step was failing (the schema was the same as jsonld above)

File: src/JsonSchema/SchemaFactory.php [hunk 2]
```diff
@@ -69,7 +69,7 @@ public function addDistinctFormat(string $format): void
      */
     public function buildSchema(string $className, string $format = 'json', string $type = Schema::TYPE_OUTPUT, ?string $operationType = null, ?string $operationName = null, ?Schema $schema = null, ?array $serializerContext = null, bool $forceCollection = false): Schema
     {
-        $schema = $schema ?? new Schema();
+        $schema = null !== $schema ? clone $schema : new Schema();
```
Comment (guilliamxavier) [inline src/JsonSchema/SchemaFactory.php:72]: This is the fix

File: src/OpenApi/Factory/OpenApiFactory.php [hunk 3]
```diff
@@ -254,9 +254,6 @@ private function collectPaths(ResourceMetadata $resourceMetadata, string $resour
                 }
             }
 
-            $schema = new Schema('openapi');
-            $schema->setDefinitions($schemas);
-
```
Comment (guilliamxavier) [inline src/OpenApi/Factory/OpenApiFactory.php:259]: Was added by #4213, not needed anymore

File: src/Swagger/Serializer/DocumentationNormalizer.php [hunk 4]
```diff
@@ -637,9 +637,7 @@ private function getJsonSchema(bool $v3, \ArrayObject $definitions, string $reso
         $schema = new Schema($v3 ? Schema::VERSION_OPENAPI : Schema::VERSION_SWAGGER);
         $schema->setDefinitions($definitions);
 
-        $this->jsonSchemaFactory->buildSchema($resourceClass, $format, $type, $operationType, $operationName, $schema, $serializerContext, $forceCollection);
-
-        return $schema;
+        return $this->jsonSchemaFactory->buildSchema($resourceClass, $format, $type, $operationType, $operationName, $schema, $serializerContext, $forceCollection);
```
Comment (guilliamxavier) [inline src/Swagger/Serializer/DocumentationNormalizer.php:640]: This is a consequence of the fix

**Review by alanpoulain — APPROVED** [review]
LGTM

File: src/Swagger/Serializer/DocumentationNormalizer.php (hunk 4 above)
Comment (guilliamxavier) [reply src/Swagger/Serializer/DocumentationNormalizer.php:640]: @soyuka: Refs https://github.com/api-platform/core/issues/4215#issuecomment-824873494. I have added a note to the changelog entry

File: src/JsonSchema/SchemaFactory.php (hunk 2 above)
Comment (soyuka) [reply src/JsonSchema/SchemaFactory.php:72]: `$schema ? clone $schema : new Schema()` would work as well right?

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,7 @@
 * Filter validation: Fix issue in Required filter validator with dot notation (#4221)
 * OpenAPI: Fix notice/warning for `response` without `content` in the `openapi_context` (#4210)
 * OpenAPI: Do not use output for request body (#4213)
+* OpenAPI: Do not use jsonld schema for all media types (#4247) (BC note: `SchemaFactory::buildSchema()` no longer modifies the passed `$schema`, callers must use the return value)
```
Comment (soyuka) [inline CHANGELOG.md:9]:
```suggestion
* OpenAPI: Do not use jsonld schema for all media types (#4247) (BC note: `SchemaFactory::buildSchema()` is now immutable as it no longer modifies the passed `$schema`)
```

File: src/JsonSchema/SchemaFactory.php (hunk 2 above)
Comment (soyuka) [inline src/JsonSchema/SchemaFactory.php:72]:
```suggestion
        $schema = $schema ? clone $schema : new Schema();
```

Comment (guilliamxavier) [reply src/JsonSchema/SchemaFactory.php]: Sorry I missed this comment. Thanks for proposing a suggestion, I've just accepted it

**Comment by alanpoulain** [comment]
Thank you @guilliamxavier.

**Pull request merged**